app/page/home/activity.tsx

```diff
@@ -51,182 +53,184 @@ const UserOpHistoryItem: React.FC<{
           </div>
-        </div>
-        {/* Transaction time */}
-        <div>
-          <div className="flex flex-col items-end">
-            <div className="mb-[4px] leading-[19px] text-[16px] text-[#000000]">
-              {creationDate}
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">
-              {creationTime}
-            </div>
+          <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
+            {tokenName}
           </div>
+          {/* Status mark */}
+          {children}
+        </div>
+        {/* To address */}
+        <div className="leading-[19px] text-[16px] text-[#bbbbbb] whitespace-nowrap">
+          to: {address.ellipsize(toAddress)}
         </div>
       </div>
-    )
-  }
+    </div>
+  )
+}
 
-  // Others' case
+const ContractLog: React.FC<{
+  toAddress: HexString
+  children?: React.ReactNode
+}> = ({ toAddress, children }) => {
   return (
     <div className="w-full flex items-center p-[14px_0px_14px_0px]">
       {/* Status and to address */}
       <div className="flex-grow flex flex-col items-start">
         {/* Status and token amount */}
         <div className="flex items-center mb-[4px]">
-          <CircleQuestion className="w-[16px] h-[16px] mr-[8px]" />
+          <ArrowRightArrowLeft className="w-[16px] h-[16px] mr-[8px]" />
           <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-            {status}
+            Interact with Dapp
           </div>
-          <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-            1.2
-          </div>
-          <div className="leading-[19px] text-[16px] text-[#000000]">ETH</div>
+          {/* Status mark */}
+          {children}
         </div>
         {/* To address */}
         <div className="leading-[19px] text-[16px] text-[#bbbbbb] whitespace-nowrap">
-          to: 0x0000...00000
+          {address.ellipsize(toAddress)}
         </div>
       </div>
-      {/* Transaction time */}
-      <div>
-        <div className="flex flex-col items-end">
-          <div className="mb-[4px] leading-[19px] text-[16px] text-[#000000]">
-            {creationDate}
-          </div>
-          <div className="leading-[19px] text-[16px] text-[#000000]">
-            {creationTime}
-          </div>
+    </div>
+  )
+}
+
+const StatusMark: React.FC<{
+  txStatus: TransactionStatus
+}> = ({ txStatus }) => {
+  if (
+    txStatus === TransactionStatus.Sent ||
+    txStatus === TransactionStatus.Pending
+  ) {
+    return (
+      <div className="p-[4px_8px_4px_8px] rounded-[4px] bg-[#F5F5F5]">
+        <div className="text-[12px] font-[500] text-[#989898]">
+          Processing...
         </div>
       </div>
+    )
+  }
+
+  if (
+    txStatus === TransactionStatus.Failed ||
+    txStatus === TransactionStatus.Reverted
+  ) {
+    return (
+      <div className="flex items-center p-[4px_8px_4px_8px] rounded-[4px] bg-[#000000]">
```

Comment:
how about `p-[4px_8px]`?

Reply:
Since the Failed icon on Figma has changed, I will not make any modifications.